crates/jstz_core/src/kv/transaction.rs

```diff
@@ -56,14 +61,34 @@ pub type Key = OwnedPath;
 struct LookupMap(BTreeMap<Key, Vec<usize>>);
 
 #[derive(Debug, Default)]
-pub struct Transaction {
+pub struct InnerTransaction {
     // A stack of transactional snapshots
     stack: Vec<Snapshot>,
     lookup_map: LookupMap,
     persistent_outbox: PersistentOutboxQueue,
     snapshot_outbox_len: u32,
 }
 
+type GuardInner = ArcMutexGuard<RawMutex, InnerTransaction>;
```

Comment:
```suggestion
type GuardInner = ArcMutexGuard<RawMutex, Pin<Box<InnerTransaction>>>;
```

This should be pinned since `Guarded` depends on the fact that pointers to `InnerTransaction` will alway be valid. Mb for missing that one

Reply:
InnerTransaction also needs a field like `_pin: PhantomPinned` to prevent rust from auto deriving `Unpin`

Reply:
yeah, unfortunately after briefly trying it out, that seems to be the case...

Reply:
ok, that seems to be right